### ISSUE TITLE
Restart tunnel, in case the portforwarding is not working

### ISSUE THREAD

**Issue by christianmerges** [t-anc/GSP-Qbittorent-Gluetun-sync-port-mod#8]
Somethines gluetun portforwarding is not working. There is a bug. Can you implement this to run this command, in case there is an issue with the portforwarding:

curl -X PUT http://localhost:8000/v1/openvpn/status \
    -H "Content-Type: application/json" \
    -d '{"status":"stopped"}'

this will stop the tunnel and it should restart after automatically. When I start after by using this command it tells me it is already running, so i guess it is not necessary. 
curl -X PUT http://localhost:8000/v1/openvpn/status \
    -H "Content-Type: application/json" \
    -d '{"status":"running"}'

**Comment by t-anc** [comment]
Hi,
I'll take a look, but I'm not sure about that. 
qdm12 is implementing a role-based authentication (https://github.com/qdm12/gluetun/issues/2238) and I'd prefer my mod to be read-only. 
Maybe I can put this restart to be optional, but honestly I feel like this is a bit out of scope.

When you say "Somethines gluetun portforwarding is not working.", what are the symptoms ? The port = 0 ? I've never had any trouble so I need a way to detect this bug. 
Is there an issue opened on gluetun's repo ? 

Thank you for your suggestion,
t-anc

**Comment by christianmerges** [comment]
yeah there is an open issue. I don't know what u mean with role based authentication, but I can run this command from the qbittorrent container without any problems and necessary auth. Your logic is already checking for port changes. It just needs to check as well after ip change, if there is a port forwarded. The bug makes the tunnel restarting and then struggles to get a port from protonvpn. I know it is more a scope of to be fixed in in gluetun, but the package is not really maintained. Maybe he is too busy with other stuff.

**Comment by t-anc** [comment]
I think I prefer to keep things simpler and limit the mod to sync the forwarded port. However if you want to make a mod to restart gluetun, feel free to take inspiration from this repo. Otherwise you could change the default healthcheck for gluetun to restart the container whenever the port is not forwarded. 
Thank you anyway for the suggestion :) 
t-anc

**Comment by qdm12** [comment]
> Is there an issue opened on gluetun's repo ?

Gluetun maintainer here, can you point me to that issue 🙏 Also what version of the program are you running?
I see https://github.com/qdm12/gluetun/issues/2297 but I'm still waiting on some feedback.

> I can run this command from the qbittorrent container without any problems and necessary auth.

FYI the auth part is not part of the latest image yet; the forwarded port route will be accessible publicly until the future v3.41 release; after that authentication will be mandatory to be setup for all routes you want an access to.

> I know it is more a scope of to be fixed in in gluetun, but the package is not really maintained. 

Hmm the fact I'm randomly reading this (closed) issue kind of contradicts this. And as @t-anc mentioned, if it's hard to re-produce, it's also hard to fix for me.

> Maybe he is too busy with other stuff.

That too let's face it 😄 But I'm still working on it 💪 !

**Comment by christianmerges** [comment]
It is here: https://github.com/qdm12/gluetun/issues/2443#issuecomment-2320767951
And you were right, a reply was missing by me. I will reply there.
